backend/src/v5/middleware/sso/aad.js

```diff
@@ -92,4 +96,30 @@ Aad.redirectToStateURL = (req, res) => {
 };
 Aad.authenticate = (redirectUri) => validateMany([addPkceProtection, authenticate(redirectUri)]);
 
+Aad.checkIfMsAccountIsLinkedTo3DRepo = async (req, res, next) => {
+	try {
+		const state = JSON.parse(req.query.state);
+
+		const { data: { id, mail } } = await getUserDetails(req.query.code, authenticateRedirectUri,
+			req.session.pkceCodes?.verifier);
+
+		try {
+			const { user, customData: { sso } } = await getUserByEmail(mail, { _id: 0, user: 1, 'customData.sso': 1 });
```

Comment:
refactor the check into models/users

This is too into the guts for a middleware to do. Just extract the data and call something like `GetUserBySSOInfo(email, ssoId)` or something